ta/pkcs11/src/processing_ec.c

```diff
@@ -624,6 +641,195 @@ enum pkcs11_rc generate_ec_keys(struct pkcs11_attribute_head *proc_params,
+	uint32_t tee_curve = 0;
+	TEE_ObjectHandle tee_obj = TEE_HANDLE_NULL;
+	TEE_Attribute tee_key_attr[1] = { };
+	TEE_Result res = TEE_ERROR_GENERIC;
+
+	if (!proc_params || !*pub_head || !*priv_head)
+		return PKCS11_CKR_TEMPLATE_INCONSISTENT;
+
+	if (remove_empty_attribute(pub_head, PKCS11_CKA_EC_POINT) ||
+		remove_empty_attribute(priv_head, PKCS11_CKA_VALUE) ||
+		remove_empty_attribute(priv_head, PKCS11_CKA_EC_PARAMS)) {
+		EMSG("Unexpected attribute(s) found");
+		trace_attributes("public-key", *pub_head);
+		trace_attributes("private-key", *priv_head);
+		return PKCS11_CKR_TEMPLATE_INCONSISTENT;
+	}
+
+	if (get_attribute_ptr(*pub_head, PKCS11_CKA_EC_PARAMS,
+						  &a_ptr, &a_size) || !a_ptr) {
+		EMSG("No EC_PARAMS attribute found in public key");
+		return PKCS11_CKR_ATTRIBUTE_TYPE_INVALID;
+	}
+
+	tee_size = ec_params2tee_keysize(a_ptr, a_size);
+	if (!tee_size)
+		return PKCS11_CKR_ATTRIBUTE_TYPE_INVALID;
+
+	tee_curve = ec_params2tee_curve(a_ptr, a_size);
+
+	TEE_InitValueAttribute(tee_key_attr, TEE_ATTR_ECC_CURVE, tee_curve, 1);
+
+	res = TEE_AllocateTransientObject(TEE_TYPE_ED25519_KEYPAIR, tee_size,
+									  &tee_obj);
+	if (res) {
+		EMSG("Transient alloc failed with %#"PRIx32, res);
+		return tee2pkcs_error(res);
+	}
+
+	res = TEE_RestrictObjectUsage1(tee_obj, TEE_USAGE_EXTRACTABLE);
+	if (res) {
+		rc = tee2pkcs_error(res);
+		goto out;
+	}
+
+	res = TEE_GenerateKey(tee_obj, tee_size, tee_key_attr, 1);
+	if (res) {
+		rc = tee2pkcs_error(res);
+		goto out;
+	}
+
+	/* Private key needs the same EC_PARAMS as used by the public key */
+	rc = add_attribute(priv_head, PKCS11_CKA_EC_PARAMS, a_ptr, a_size);
+	if (rc)
+		goto out;
+
+	rc = tee2pkcs_add_attribute(priv_head, PKCS11_CKA_VALUE,
+								tee_obj, TEE_ATTR_ED25519_PRIVATE_VALUE);
+	if (rc)
+		return rc;
+
+	rc = tee2pkcs_add_attribute(priv_head, PKCS11_CKA_EC_POINT,
+								tee_obj, TEE_ATTR_ED25519_PUBLIC_VALUE);
+	if (rc)
+		return rc;
+
+	rc = tee2pkcs_add_attribute(pub_head, PKCS11_CKA_EC_POINT,
+								tee_obj, TEE_ATTR_ED25519_PUBLIC_VALUE);
+	if (rc)
+		return rc;
+
+	out:
+	if (tee_obj != TEE_HANDLE_NULL)
+		TEE_CloseObject(tee_obj);
+
+	return rc;
+}
+
+enum pkcs11_rc
+pkcs2tee_proc_params_eddsa(struct active_processing *proc,
+				  struct pkcs11_attribute_head *proc_params)
+{
+	enum pkcs11_rc rc = PKCS11_CKR_GENERAL_ERROR;
+	uint32_t flag = 0, ctx_len = 0;
+	void *ctx_data = NULL;
+	struct serialargs args = { };
+	struct eddsa_processing_ctx *ctx = NULL;
+
+	serialargs_init(&args, proc_params->data, proc_params->size);
+
+	rc = serialargs_get_u32(&args, &flag);
+	if (rc)
+		return rc;
+
+	rc = serialargs_get_u32(&args, &ctx_len);
+	if (rc)
+		return rc;
+	
+	rc = serialargs_get_ptr(&args, &ctx_data, ctx_len);
+	if (rc)
+		return rc;
+
+	if (serialargs_remaining_bytes(&args))
+		return PKCS11_CKR_ARGUMENTS_BAD;
+
+	proc->extra_ctx = TEE_Malloc(sizeof(struct eddsa_processing_ctx) +
+								 ctx_len,
+								 TEE_USER_MEM_HINT_NO_FILL_ZERO);
```

Comment:
```patch
 	proc->extra_ctx = TEE_Malloc(sizeof(struct eddsa_processing_ctx) +
-								 ctx_len,
-								 TEE_USER_MEM_HINT_NO_FILL_ZERO);
+				      ctx_len, TEE_USER_MEM_HINT_NO_FILL_ZERO);

Reply:
Fixed indentation,
Is it Ok? 
sizeof(*ctx) + ctx_len,